File: frontend/src/models/cnpj.js
export default class CNPJ {

  regexCNPJ = /^\d{2}.\d{3}.\d{3}\/\d{4}-\d{2}$/;

  constructor(cnpj) {
    this.cnpj = cnpj;
  }

  isValid() {
    return this.regexCNPJ.test(this.cnpj);
  }

  format() {
    this.cnpj = this.cnpj.replace(/\D/g, "");
    this.cnpj = this.cnpj.replace(/^(\d{2})(\d)/, "$1.$2");
    this.cnpj = this.cnpj.replace(/^(\d{2})\.(\d{3})(\d)/, "$1.$2.$3");
    this.cnpj = this.cnpj.replace(/\.(\d{3})(\d)/, ".$1/$2");
    this.cnpj = this.cnpj.replace(/(\d{4})(\d)/, "$1-$2");
  }
}

File: backend/main_app/app/models/cnpj.py
import re


class CNPJ:
    value = None
    __format = '^[0-9]{2}\.[0-9]{3}\.[0-9]{3}\/[0-9]{4}-[0-9]{2}$'

    def __init__(self, cnpj: str):
        self.value = cnpj

    def validate(self) -> bool:
        if self.value is None:
            return False
        else:
            return re.search(self.__format, self.value) is not None
